build/plugins/hypothesis.html

```diff
@@ -13,17 +23,83 @@
             }
         };
     };
-</script>
 
-<!-- annotations plugin -->
+    // hypothesis client script
+    const embed = 'https://hypothes.is/embed.js';
+    // hypothesis annotation count query url
+    const query = 'https://api.hypothes.is/api/search?limit=0&url='
 
-<script src='https://hypothes.is/embed.js'>
-    // /////////////////////////
-    // DESCRIPTION
-    // /////////////////////////
+    
+    // start script
+    function start() {
+        const button = makeButton();
+        document.body.insertBefore(button, document.body.firstChild);
+        insertCount(button);
+    }
 
-    // This third-party plugin 'Hypothesis' allows public annotation of the
-    // manuscript.
+    // make button
+    function makeButton() {
+        // create button
+        const button = document.createElement('button');
+        button.id = 'hypothesis_button';
+        button.innerHTML = document.querySelector(
+            '.icon_hypothesis'
+        ).innerHTML;
+        button.title = 'Hypothesis annotations';
+        button.classList.add('icon_button');
 
-    // https://web.hypothes.is/
+        function onClick(event) {
+            onButtonClick(event, button);
+        }
+
+        // attach click listeners
+        button.addEventListener('click', onClick);
+
+        return button;
+    }
+
+    // insert annotations count
+    async function insertCount(button) {
+        // get annotation count from Hypothesis based on url
+        let count = '-';
+        try {
+            const canonical = document.querySelector('link[rel="canonical"]');
+            const location = window.location;
+            const response = await fetch(query + (canonical || location).href);
```

Comment:
I'm not sure whether we need to escape the `url` / `uri` parameter.

Reply:
Are you asking "do we need to escape the url parameter", because I am not doing that here. Theoretically I think the browser should escape it automatically when making the query.

Reply:
If you're talking about me using `canonical`, that is so Hypothesis will return some results when you are looking at a local manuscript. Because it wont return (correct) annotation counts for `localhost:3000/`, `C:/users/vincent/test-manuscript.html`, etc.

Reply:
using canonical makes sense to me.

My question was whether it needs to be encoded like `encodeURIComponent(canonical)`. It's never clear to me when query string parameters need to be encoded like this.

Reply:
To be honest, it's not clear to me either. I think it might depend on the code reading the info on the other end too? Like does it use `decodeURIComponent`...

I think the browser should handle it in most cases though. Consider this contrived url with a space in it: `https://greenelab.github.io/xswap manuscript/`

If you do `fetch('https://greenelab.github.io/xswap manuscript/')`, it has no problem properly encoding the space (not a valid url character) and completing the request (although the request returns a 404 because it's not a real url).

If you put that url into the location bar in your browser, it also properly converts it.

Reply:
> I think it might depend on the code reading the info on the other end too? Like does it use decodeURIComponent...

I think it probably does because elsewhere in [docs](https://h.readthedocs.io/en/latest/api-reference/v1/)

>  ![image](https://user-images.githubusercontent.com/1117703/83831827-3021c680-a6b6-11ea-99c4-c8b1189fc06b.png)

If you try the following URL, we still get 122 returned:
https://api.hypothes.is/api/search?limit=0&url=https%3A%2F%2Flelaboratoire.github.io%2Frethink-prs-ms%2F

So I think we should use `encodeURIComponent`... it's probably a bit safer by avoiding some cases where the API truncates the `url` value because of an unescaped character... although I might be missing something.